pkg/strategy/dca2/recover.go

```diff
@@ -0,0 +1,266 @@
+package dca2
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/core"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type queryAPI interface {
+	QueryOpenOrders(ctx context.Context, symbol string) ([]types.Order, error)
+	QueryClosedOrdersDesc(ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64) ([]types.Order, error)
```

Comment:
I think you need to pull out this method QueryClosedOrdersDesc to a single interface, like descendingClosedOrderQueryService interface { ... }